op-plasma/daclient.go

```diff
@@ -0,0 +1,93 @@
+package plasma
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// ErrNotFound is returned when the server could not find the input.
+var ErrNotFound = errors.New("not found")
+
+// ErrCommitmentMismatch is returned when the server returns the wrong input for the given commitment.
+var ErrCommitmentMismatch = errors.New("commitment mismatch")
+
+// ErrInvalidInput is returned when the input is not valid for posting to the DA storage.
+var ErrInvalidInput = errors.New("invalid input")
+
+// DAClient is an HTTP client to communicate with a DA storage service.
+// It creates commitments and retrieves input data + verifies if needed.
+// Currently only supports Keccak256 commitments but may be extended eventually.
+type DAClient struct {
+	url    string
+	verify bool
+}
+
+func NewDAClient(url string) *DAClient {
```

Comment:
This should take `VerifyOnRead` because it is a constructor argument that is basically never set later on in the lifecycle of the DA Client.